src/datastore/postgres-store.ts

```diff
@@ -4806,7 +4806,7 @@ export class PgDataStore
       `
       SELECT sum(fee_rate) as fee_sum
       FROM txs
-      WHERE canonical = true AND microblock_canonical = true AND sender_address = $1 AND block_height <= $2
+      WHERE canonical = true AND microblock_canonical = true AND (sender_address = $1 OR sponsor_address = $1) AND block_height <= $2
```

Comment:
Is there a unit test that replicates the negative balance bug in https://github.com/hirosystems/stacks-blockchain-api/issues/857? Won't this query include fees for a sponsored tx for both the sender and the sponsor?

Reply:
> Won't this query include fees for a sponsored tx for both the sender and the sponsor?

hmm, I think you're right. 

> Is there a unit test that replicates the negative balance bug in #857?
Not yet, I'm working on it.

Reply:
Can you include an explicit unit test for the negative balance behavior created by the situation in the referenced issue?

Reply:
Done!